pandas/core/frame.py

```diff
@@ -1954,19 +1956,23 @@ def to_stata(
                 raise ValueError("strl is not supported in format 114")
             from pandas.io.stata import StataWriter as statawriter
         elif version == 117:
-            from pandas.io.stata import StataWriter117 as statawriter
+            # mypy: Name 'statawriter' already defined (possibly by an import)
+            from pandas.io.stata import StataWriter117 as statawriter  # type: ignore
         else:  # versions 118 and 119
-            from pandas.io.stata import StataWriterUTF8 as statawriter
+            # mypy: Name 'statawriter' already defined (possibly by an import)
+            from pandas.io.stata import StataWriterUTF8 as statawriter  # type:ignore
 
         kwargs = {}
         if version is None or version >= 117:
             # strl conversion is only supported >= 117
             kwargs["convert_strl"] = convert_strl
         if version is None or version >= 118:
             # Specifying the version is only supported for UTF8 (118 or 119)
-            kwargs["version"] = version
+            # mypy: Incompatible types in assignment
+            kwargs["version"] = version  # type: ignore
 
-        writer = statawriter(
+        # mypy: Too many arguments for "StataWriter"
+        writer = statawriter(  # type: ignore
```

Comment:
could lose this also.. (not sure about this though @WillAyd ?)

```diff
diff --git a/pandas/core/frame.py b/pandas/core/frame.py
index 36a1f8747..98dcb3a0e 100644
--- a/pandas/core/frame.py
+++ b/pandas/core/frame.py
@@ -1971,8 +1971,7 @@ class DataFrame(NDFrame):
             # mypy: Incompatible types in assignment
             kwargs["version"] = version  # type: ignore
 
-        # mypy: Too many arguments for "StataWriter"
-        writer = statawriter(  # type: ignore
+        writer = statawriter(
             path,
             self,
             convert_dates=convert_dates,
diff --git a/pandas/io/stata.py b/pandas/io/stata.py
index cb446b0af..50b8e2934 100644
--- a/pandas/io/stata.py
+++ b/pandas/io/stata.py
@@ -2103,6 +2103,7 @@ class StataWriter(StataParser):
         time_stamp: Optional[datetime.datetime] = None,
         data_label: Optional[str] = None,
         variable_labels: Optional[Dict[Label, str]] = None,
+        **kwargs,  # dummy parameter for compat with StataWriter117
     ):
         super().__init__()
         self._convert_dates = {} if convert_dates is None else convert_dates
```

Reply:
Not a strong preference but I slightly prefer ignore with explicit arguments instead of kwargs. Emphasis on not a strong preference if you feel otherwise

Reply:
fair enough. no strong preference either. so ignore is fine.

Reply:
I also don't like extraneous **kwargs that just disappear.

In hindsight, I would prefer that StataWriter, StataWriter117, and StataWriterUTF8 all have the same signature on all public facing functions and constructors. This would mean that they would all really be "StataWriters" and this problem would be gone.